### PR TITLE
Add details about DNS lookup API

### PR DESCRIPTION
**Pull request by lidel**
This PR aims to document #10 
I am quite sure it needs proof-reading.

(Not sure how I ended up writing this, I was just going to warm up before writing the Origin one :thinking:)

### CODE REVIEW

File: content/dns-lookup-api.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+title: "DNS lookup API"
+date: 2018-03-29T14:49:22+02:00
+---
+
+An API that allows a WebExtension to lookup for arbitrary DNS records in highly
+efficient manner.
```
Comment (olizilla) [inline content/dns-lookup-api.md]: > An API that allows a WebExtension to lookup for arbitrary DNS records in highly efficient manner.

This is a tricky sentence, as you'd normal _allow a WebExtension to **look up** a **thing**_, without the ~~for~~ in there, but for the special case of dns _lookups_  you'd say something more specific like

_An API that allows a WebExtension to perform DNS lookups for arbitrary record types in a highly efficient manner._

File: content/dns-lookup-api.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+There are two DNS resource record types that are relevant to this objective:
+
+- `SRV` records provide means for specifying the location of services ([RFC 2782][2]).
+- `TXT` records provide the ability to associate arbitrary string metadata with a host ([RFC 1464](https://tools.ietf.org/html/rfc1464)).
+
+`TXT` records are much more versatile. There are historical examples of
+[publishing PGP pubkeys][3], [domain ownership verification][4],
+['serverless' redirect services][5] and more.
+
+IPFS uses `TXT` record for publishing [dnslink][6] as a means of exposing content
```
Comment (olizilla) [inline content/dns-lookup-api.md]: IPFS uses `TXT` record**s**

File: content/dns-lookup-api.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+from IPFS Path under `/ipns/${fqdn}/` namespace. Validation of `/ipns/` paths
```
Comment (olizilla) [inline content/dns-lookup-api.md]: from `an` IPFS Path

File: content/dns-lookup-api.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+includes DNS lookup to verify if `/ipns/${fqdn}` is backed by the presence of
```
Comment (olizilla) [inline content/dns-lookup-api.md]: includes `a` DNS lookup

File: content/dns-lookup-api.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+dnslink `TXT` record.
```
Comment (olizilla) [inline content/dns-lookup-api.md]: `a` dnslink

File: content/dns-lookup-api.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+`TXT` records are much more versatile. There are historical examples of
+[publishing PGP pubkeys][3], [domain ownership verification][4],
+['serverless' redirect services][5] and more.
+
+IPFS uses `TXT` record for publishing [dnslink][6] as a means of exposing content
+from IPFS Path under `/ipns/${fqdn}/` namespace. Validation of `/ipns/` paths
+includes DNS lookup to verify if `/ipns/${fqdn}` is backed by the presence of
+dnslink `TXT` record. 
+
+Without dedicated API for DNS lookups browser extensions are forced to use
```
Comment (olizilla) [inline content/dns-lookup-api.md]: Without `a` dedicated API

File: content/dns-lookup-api.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+third-party [DNS-over-HTTPS][7] services.
+This workaround comes at a price:
+
+- Dependency on hardcoded third party lookup service introduces the single
```
Comment (olizilla) [inline content/dns-lookup-api.md]: introduces ~~the~~ `a` single point of failure

File: content/dns-lookup-api.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  point of failure.  It also makes MITM attacks easier and increases
+  probability of leaking private information.
+
+- Sending HTTP GET for each query is much slower than native DNS client already
```
Comment (olizilla) [inline content/dns-lookup-api.md]: Sending `an` HTTP GET

File: content/dns-lookup-api.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+ present in web browser. Overhead makes it not feasible to run
+ `TXT` lookup too often or during time-critical paths such as
```
Comment (olizilla) [inline content/dns-lookup-api.md]: Overhead makes it not feasible to run `TXT` lookup too often or during time-critical paths...
=>
The overhead is particularly undesirable during time-critical paths...

File: content/dns-lookup-api.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+ [blocking `onBeforeRequest` handler][8] (degrades browsing performance, kills battery).
+
+
+
+## Usage Documentation
+
+Recently added [`browser.dns.resolve`][1] API from Firefox 60 is a good starting point.
```
Comment (olizilla) [inline content/dns-lookup-api.md]: `The` recently added

File: content/dns-lookup-api.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+Example below shows how lookup for `A` record works:
```
Comment (olizilla) [inline content/dns-lookup-api.md]: `The` example below

File: content/dns-lookup-api.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+let resolving = browser.dns.resolve("example.com");
+resolving.then(resolved);
+
+// > e.g. Array [ "73.284.240.12" ]
+```
+
+### What Is Missing
+
+There should be an additional parameter that enables WebExtension to lookup for
+record types different from the default `A`, namely `TXT`.
```
Comment (olizilla) [inline content/dns-lookup-api.md]: I'd merge the what is missing section into the **Usage Documentation** section and show an example the api we'd like to see.

**Comment by lidel** [comment]
Thanks! Updated version: [content/dns-lookup-api.md](https://github.com/ipfs-shipyard/arewedistributedyet/blob/632310206734f912428feb60d51490f31360a55e/content/dns-lookup-api.md)

**Pull request merged**